cmd/lakectl/cmd/retry_client_test.go

```diff
@@ -0,0 +1,236 @@
+			expectedShouldRetry: true,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 429 Too Many Requests",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusTooManyRequests,
+			},
+			err:                 nil,
+			expectedShouldRetry: true,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 500 Internal Server Error",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusInternalServerError,
+			},
+			err:                 nil,
+			expectedShouldRetry: true,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 503 Service Unavailable",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusServiceUnavailable,
+			},
+			err:                 nil,
+			expectedShouldRetry: true,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 401 Unauthorized",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusUnauthorized,
+			},
+			err:                 nil,
+			expectedShouldRetry: false,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 404 Not Found",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusNotFound,
+			},
+			err:                 nil,
+			expectedShouldRetry: false,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 200 Ok",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusOK,
+			},
+			err:                 nil,
+			expectedShouldRetry: false,
+			expectedError:       "",
+		},
+		{
+			name: "HTTP Status - 201 Created",
+			getTestContext: func() context.Context {
+				return context.Background()
+			},
+			resp: &http.Response{
+				StatusCode: http.StatusCreated,
+			},
+			err:                 nil,
+			expectedShouldRetry: false,
+			expectedError:       "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// if tc.err != nil {
+			// 	errStr := tc.err.Error()
+			// 	fmt.Sprintf("%s", errStr)
+			// }
+			shouldRetry, err := lakectlRetryPolicy(tc.getTestContext(), tc.resp, tc.err)
+			require.Equal(t, tc.expectedShouldRetry, shouldRetry)
+			if tc.expectedError != "" {
+				require.EqualError(t, err, tc.expectedError)
+			}
+		})
+	}
+}
+
+func TestRetryHTTPClient(t *testing.T) {
+	retryCount := -1
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.MaxIdleConnsPerHost = DefaultMaxIdleConnsPerHost
+	retryClient := NewRetryClient(10, 10*time.Millisecond, 30*time.Millisecond, transport)
```

Comment:
Can we somehow mock the client used in the retry client instead of the server and avoid the usage of channels?

Reply:
@arielshaqed mentioned something similar. Before we get into improving the test, please see [this](https://github.com/treeverse/lakeFS/pull/7714#discussion_r1587207149) comment and [this](https://github.com/treeverse/lakeFS/pull/7714#discussion_r1587246163) comment. Let me know what you think. If we decide to keep these tests, we can discuss how to improve them.

Reply:
IMO, testing the custom `lakectlRetryPolicy` is enough here
Don't know if we should test the retry mechanism here, but adding some tests that lakectl does retries as expected when configured with retries would be helpful